### PR TITLE
feat(audits/server): Test that `null` is allowed for body parameters

### LINKED ISSUE #26: Should have audits for accepting `null`s
The spec has a paragraph that states:

> Note: Specifying `null` in JSON (or equivalent values in other formats) as values for optional request parameters is equivalent to not specifying them at all.

The audit suite should test this explicitly.

(I was inspired by the audit suite to be stricter about banning things like `variables: 0` and got carried away and also banned `variables: null`. So the suite should prevent this over-fitting.)

### PR DESCRIPTION
**Pull request by glasser**
As per spec:

> Note: Specifying `null` in JSON (or equivalent values in other
> formats) as values for optional request parameters is equivalent to
> not specifying them at all.

Fixes #26.

### CODE REVIEW

File: src/audits/server.ts [hunk 1]
```diff
@@ -445,6 +445,51 @@ export function serverAudits(opts: ServerAuditOptions): Audit[] {
...
+          });
+          assert('Status code', res.status).toBe(200);
+          assert(
+            'Execution result',
+            await assertBodyAsExecutionResult(res),
+          ).notToHaveProperty('errors');
+        },
+      ),
+      audit(
+        'MUST allow null {${parameter}} parameter when accepting application/json',
```
Comment (enisdenjo) [inline src/audits/server.ts:472]:
```suggestion
        `MUST allow null {${parameter}} parameter when accepting application/json`,
```

**Review by enisdenjo — CHANGES_REQUESTED** [review]

File: src/audits/server.ts (hunk 1 above)
Comment (glasser) [reply src/audits/server.ts:472]: yep already done

**Comment by glasser** [comment]
Note: The `apollo-server` implementation directly currently installs AS v4.1, which passes these new audits. v4.2 fails it (an accidental regression ironically created while trying to improve the set of SHOULDs passed by AS). After this PR is released I'll put out a new v4.2.1 which will pass (which would be a great time to upgrade and rerun the version pinned in this repo).

**Comment by glasser** [comment]
Closing in favor of identical #28 from this repository (so the Report workflow works)

**Pull request closed without merging**